### ISSUE TITLE
Error: File not found (/node_modules/...)

### ISSUE THREAD

**Issue by NatoBoram** [firefox-devtools/vscode-firefox-debug#150]
I have followed the tutorial for [Chrome Debugging with Angular CLI](https://github.com/microsoft/vscode-recipes/tree/master/Angular-CLI) and created a configuration of type `firefox`.

```json
{
  "name": "Debug on Firefox",
  "type": "firefox",
  "request": "launch",
  "preLaunchTask": "npm: start",
  "url": "https://localhost:4200/",
  "webRoot": "${workspaceFolder}"
}
```

However, when I try to launch it, there's a bunch of errors.

> Unable to open 'check-correctness-of-iteration.js': Unable to read file (Error: File not found (/node_modules/@angular-devkit/build-angular/node_modules/core-js/internals/check-correctness-of-iteration.js)).

> Unable to open 'sockjs.js': Unable to read file (Error: File not found (/node_modules/sockjs-client/dist/sockjs.js)).

> Unable to open 'router.js': Unable to read file (Error: File not found (/node_modules/@angular/router/fesm5/router.js)).

If I just press "Continue" until there's no more errors, then the app works well... until I navigate. Then I have more errors.

> Unable to open 'router.js': Unable to read file (Error: File not found (/node_modules/@angular/router/fesm5/router.js)).

I suspect there's a relation with https://github.com/firefox-devtools/vscode-firefox-debug/issues/140. If there's a solution, it would be great to add it to https://github.com/microsoft/vscode-recipes/issues/235 and https://github.com/johnpapa/vscode-angular-snippets/issues/99.

**Comment by hbenl** [comment]
This is a combination of 2 issues:
- the debugger breaks on all exceptions (even those that are caught) due to Microsoft/vscode#84295. As a workaround you can toggle one of the exception breakpoint checkboxes in the UI while debugging, then VS Code will tell this extension about your exception breakpoint settings (but you'll have to do this at the start of *every* debugging session...)
- there is a bug in this extension that is currently making it impossible to step into some bundled sources (that's why it says "Unable to open ..."). I'm working on it.

**Comment by hbenl** [comment]
FYI: in version 2.5.0 I added a workaround to reduce the impact of the first issue: by default, the debugger will only break on *uncaught* exceptions (until the user toggles one of the exception breakpoint checkboxes).

**Comment by hbenl** [comment]
The second issue can be fixed by adding the following pathMappings to your configuration:
```
{ "url": "webpack:///node_modules/", "path": "${webRoot}/node_modules/" },
{ "url": "webpack:///webpack", "path": null },
{ "url": "webpack:///(webpack)", "path": null }
```

I will add these as defaults in the next release.

**Comment by hbenl** [comment]
Both issues are fixed in version 2.5.1
